app/ui/legacy/src/main/java/com/fsck/k9/ui/settings/SettingsListFragment.kt

```diff
@@ -239,10 +238,20 @@ class SettingsListFragment : Fragment(), ItemTouchCallback {
     }
 
     override fun itemTouchDropped(oldPosition: Int, newPosition: Int) {
-        // Save the new item order, e.g. in your database
+        // Persist the new account order
         val preferences = Preferences.getPreferences(context)
         val moveUp = oldPosition <= newPosition
-        val accountItem = itemAdapter.getAdapterItem(oldPosition) as AccountItem
-        preferences.move(accountItem.account, moveUp)
+        var rowsToMove = abs(newPosition - oldPosition)
+        var prevPos = oldPosition
+        while (rowsToMove > 0){
+            val accountItem = itemAdapter.getAdapterItem(prevPos) as AccountItem
+            preferences.move(accountItem.account, moveUp)
+            if (moveUp){
+                prevPos++
+            } else {
+                prevPos--
+            }
+            rowsToMove--
+        }
     }
```

Comment:
I guess you proved my statement "`Preferences.move()` needs to be modified to allow moving an account to any position" false. However, `Preferences.move()` *should* be modified so this code can be written as:

```kotlin
override fun itemTouchDropped(oldPosition: Int, newPosition: Int) {
    if (oldPosition == newPosition) return

    val account = (itemAdapter.getAdapterItem(newPosition) as AccountItem).account
    val firstAccountPosition = itemAdapter.adapterItems.indexOfFirst { it is AccountItem }
    val newAccountPosition = newPosition - firstAccountPosition

    // Persist the new account order
    val preferences = Preferences.getPreferences(context)
    preferences.move(account, newAccountPosition)
}
```

In order to do that you can change `AccountPreferenceSerializer.move()` (that is called from `Preferences.move()`) to this:

```kotlin
fun move(editor: StorageEditor, account: Account, storage: Storage, newPosition: Int) {
    val accountUuids = storage.getString("accountUuids", "").split(",").filter { it.isNotEmpty() }

    val oldPosition = accountUuids.indexOf(account.uuid)
    if (oldPosition == -1 || oldPosition == newPosition) return

    val newAccountUuidsString = accountUuids.toMutableList()
        .apply {
            removeAt(oldPosition)
            add(newPosition, account.uuid)
        }
        .joinToString(separator = ",")

    editor.putString("accountUuids", newAccountUuidsString)
}
```